### PR TITLE
Unify data processing

### LINKED ISSUE #74: [demo] Unify data processing for demo projects
Unify the data processing for movieLens-1m, movielens-25m, criteo-5d and other datasets,  including feature generation, match dataset generation, ranking dataset generation, negative sampling, etc.

### PR DESCRIPTION
**Pull request by intelligencegear**
Unify the data processing for movieLens-1m, movielens-25m, criteo-5d and other datasets, including feature generation, match dataset generation, ranking dataset generation, negative sampling, etc.  

In addition, we optimize feature generation and negative sampling processing code that can run on MovieLens-1M/25M dataset to process feature generation, matching dataset generation, ranking dataset generation more quickly now. 

close #74.

### CODE REVIEW

File: demo/dataset/README-CN.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    value = -1
+    try:
+        if x is not None:
+            value = float(x)
+    except ValueError:
+        pass
+    return int(np.floor(np.log(value) ** 2)) if value>2.0 else int(value)
+```
+
+由于原始数据较大，我们只是验证 CTR 模型，这里选取了训练集前 5 天对样本，测试集第 1 天的样本进行验证。假设我们位于dataset项目的根目录，我们可以通过执行以下命令准备好 CTR 模型训练所使用的样本数据：
```
Comment (wikty) [inline demo/dataset/README-CN.md:93]: a little confuse about this sentence

File: demo/dataset/README-CN.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+```shell
+python rank_dataset.py --conf rank.yaml.dev --verbose
+```
+
+## Criteo-5D
+在这一节，我们使用由 CriteoLabs 公开的广告曝光点击日志 [Terabyte Click Logs](https://labs.criteo.com/2013/12/download-terabyte-click-logs-2/) 来进行演示。
+
+```python
+import metaspore
+metaspore.demo.download_dataset()
```
Comment (wikty) [inline demo/dataset/README-CN.md:76]: Does this method need a dataset name?

File: demo/dataset/criteo/fg.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+import sys
+import yaml
+import time
+import argparse
+import subprocess
+
+from pyspark.sql import SparkSession
+
+sys.path.append('../')
```
Comment (wikty) [inline demo/dataset/criteo/fg.py:25]: Don't find a `__init__.py` file in the `common` package, does it work?

File: demo/dataset/README.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+```
+
+## Criteo-5D
+In this section, we use the publicly available dataset [Terabyte Click Logs](https://labs.criteo.com/2013/12/download-terabyte-click-logs-2/) published by CriteoLabs as our demo dataset. 
+```python
+import metaspore
+metaspore.demo.download_dataset()
+```
+
+If the downloading fails, please refer to [MetaSpore Demo Dataset](https://ks3-cn-beijing.ksyuncs.com/dmetasoul-bucket/demo/criteo/index.html) and download the dataset manually. We normalize numerical values by transforming from a value z to log(z) if z > 2, which is proposed by the winner of [Display Advertising Challenge](https://www.kaggle.com/c/criteo-display-ad-challenge) in [3 Idiots' Approach](https://github.com/ycjuan/kaggle-2014-criteo).
```
Comment (wikty) [inline demo/dataset/README.md:82]: The `log(z) if z > 2` isn't same with the following code.

File: demo/dataset/README-CN.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+```
+
+数据集比较大，如果您下载失败，可以重试或者通过 [MetaSpore Demo Dataset](https://ks3-cn-beijing.ksyuncs.com/dmetasoul-bucket/demo/criteo/index.html) 来进行手动下载。 根据2014年 [Display Advertising Challenge](https://www.kaggle.com/c/criteo-display-ad-challenge) 优胜团队提供的方法 [3 Idiots' Approach](https://github.com/ycjuan/kaggle-2014-criteo) ，我们对里面对数值特征进行离散处理：
```
Comment (wikty) [inline demo/dataset/README-CN.md:79]: typos: maybe change "对里面对数值特征" to "对里面的数值特征"

File: demo/dataset/README-CN.md [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+### 1. 特征生成
+假设我们位于dataset项目的根目录，我们可以通过执行以下命令来完成特征生成的工作：
+
+ ```shell
+ cd ml_1m
+ python fg.py --conf fg.yaml.dev --verbose
+ ```
+
+### 2. 召回样本生成
+假设我们位于dataset项目的根目录，我们可以通过执行以下命令准备好 `Collaborative Filtering` 算法训练所使用的样本数据：
```
Comment (wikty) [inline demo/dataset/README-CN.md:26]: It's better give a link to the corresponding algo/model(CF/TwoTowers/CTR) training code. Users can better relate the dataset processing with model training.

File: demo/dataset/README.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
+## [中文介绍](README-CN.md)
+
+# Data Processing and Preparation
+In this project, data processing is unified for MovieLens-1M, MovieLens-25M, Criteo-5d and other datasets, including feature generation,  matching dataset generation, ranking dataset generation, negative sampling, etc.  If you are Chinese developer, you may like to visit our [CN Doc](README-CN.md).
```
Comment (wikty) [inline demo/dataset/README.md:4]: The following sentence is useless.

> If you are Chinese developer, you may like to visit our [CN Doc](README-CN.md).

File: demo/dataset/README-CN.md (hunk 2 above)
Comment (intelligencegear) [reply demo/dataset/README-CN.md:76]: No, please refer to: https://github.com/meta-soul/MetaSpore/blob/main/tutorials/metaspore-getting-started.ipynb

File: demo/dataset/criteo/fg.py (hunk 3 above)
Comment (intelligencegear) [reply demo/dataset/criteo/fg.py:25]: It works. Please refer to this [link](https://stackoverflow.com/questions/37139786/is-init-py-not-required-for-packages-in-python-3-3)

File: demo/dataset/README.md (hunk 4 above)
Comment (intelligencegear) [reply demo/dataset/README.md:82]: Change it to: We normalize numerical values by transforming from a value `z` to `floor(log(z)^2)` if `z > 2`

File: demo/dataset/README-CN.md (hunk 6 above)
Comment (intelligencegear) [reply demo/dataset/README-CN.md:26]: I fix that for CTR models now.

File: demo/dataset/README-CN.md (hunk 1 above)
Comment (intelligencegear) [reply demo/dataset/README-CN.md:93]: Remove "由于原始数据较大，我们只是验证 CTR 模型".

File: demo/dataset/criteo/fg.py (hunk 3 above)
Comment (intelligencegear) [reply demo/dataset/criteo/fg.py:25]: I will add an empty file in this directory.

**Review by dmetasoul01 — APPROVED** [review]

**Pull request merged**